src/zarr/core/group.py

```diff
@@ -2225,120 +2199,82 @@ def create_array(
         name: str,
         *,
         shape: ShapeLike,
-        dtype: npt.DTypeLike = "float64",
-        fill_value: Any | None = None,
+        dtype: npt.DTypeLike,
+        chunks: ChunkCoords | Literal["auto"] = "auto",
+        shards: ChunkCoords | None = None,
+        filters: FiltersParam = "auto",
+        compressors: CompressorsParam = "auto",
```

Comment:
For backward compatibility, we should also support the `compressor` argument. The logic should be simple though:

```python
if compressor:
    warnings.warn("compressor is deprecated ...")
    compressors = (compressor, )
```

Reply:
Should we do that just in `(Async)Group.create_array` or also in the new top-level `zarr.api.(a)synchronous.create_array` methods?

Reply:
my feeling is that the brand new functions shouldn't have to worry about deprecated APIs

Reply:
Added in https://github.com/zarr-developers/zarr-python/pull/2463/commits/be60d73ce00f0ed9a23cf8bd4e3e80fd96f58056